Client/Frontend/Browser/PasswordHelper.swift

```diff
@@ -171,4 +172,95 @@ class PasswordHelper: BrowserHelper {
             })
         })
     }
+
+    func handleAuthRequest(viewController: UIViewController, challenge: NSURLAuthenticationChallenge, completion: (password: Password?) -> Void) {
+        // If there have already been too many login attempts, we'll just fail.
+        if challenge.previousFailureCount > PasswordHelper.MaxAuthenticationAttempts {
```

Comment:
Maybe change this to `>=`, then bump `MaxAuthenticationAttempts` to 3? Should be the same behavior, but makes it clear that there are 3 tries, not 2.